Content.Shared/_RMC14/Xenonids/Construction/SharedXenoHiveCoreSystem.cs

```diff
@@ -1,3 +1,30 @@
+using Content.Shared._RMC14.Xenonids.Announce;
+using Content.Shared._RMC14.Xenonids.Hive;
+
 ﻿namespace Content.Shared._RMC14.Xenonids.Construction;
 
-public abstract class SharedXenoHiveCoreSystem : EntitySystem;
+public abstract class SharedXenoHiveCoreSystem : EntitySystem
+{
+    [Dependency] private readonly SharedXenoAnnounceSystem _announce = default!;
+    [Dependency] private readonly SharedXenoHiveSystem _hive = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<HiveCoreComponent, ComponentShutdown>(OnShutdown);
```

Comment:
For this it might be better to use DestructionEventArgs so it doesn't fire when admin-erased or at round-end

Reply:
already shouldnt happen on round end since GetHive checks if the hive is terminating, admins deleting it would brick the hive and i think vv cant edit the limits for sone reason, so you would have no way to fix it